vortex-array/src/arrays/primitive/compute/take.rs

```diff
@@ -17,6 +23,27 @@ impl TakeFn<&PrimitiveArray> for PrimitiveEncoding {
         let indices = indices.to_primitive()?;
         let validity = array.validity().take(&indices)?;
 
+        if array.ptype() != PType::F16
```

Comment:
There's a FLUP here which is to re-interpret cast everything to unsigned integers so we don't have to exclude F16.